File: AddressBook/AddressBook/Dtos/PersonLabel.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Entities;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class PersonLabelRes
    {
        public int PersonLabelId { get; set; }
        public int? LabelId { get; set; }
        public LabelRes? Label { get; set; }
        public int? PersonId { get; set; }
        public PersonRes? Person { get; set; }
    }

    public class PersonLabelReqEdit
    {
        public int? PersonId { get; set; }
        public int? LabelId { get; set; }
    }

    public class PersonLabelReqSearch : PagedReq
    {
        [Required]
        public int? PersonId { get; set; }
    }
}

File: AddressBook/react-client/src/dtos/PersonLabel.ts
import Common from "../utility/Common";
import { LabelRes } from "./Label";
import { PagedReq } from "./PagedReq";
import { PersonRes } from "./Person";

export interface PersonLabelRes {
  personLabelId?: number;
  labelId?: number;
  label?: LabelRes;
  personId?: number;
  person?: PersonRes;
}

export class PersonLabelReqEdit {
  labelId?: number = 0;
  personId?: number = 0;
}

export class PersonLabelReqSearch extends PagedReq {
  personId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {personId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.personId = personId;
  }
}